src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -1,22 +1,70 @@
 package frc.robot.subsystems;
 
-import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
 import edu.wpi.first.wpilibj.Servo;
-import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.PIDSubsystem;
 import frc.robot.Constants;
 
+
 /**
  * Creates subsystem and functions for the shooter.
  */
-public class Shooter extends SubsystemBase {
-    private final WPI_TalonSRX shooter = new WPI_TalonSRX(Constants.Motors.shooterID);
+public class Shooter extends PIDSubsystem {
+    private final WPI_TalonFX shooter = new WPI_TalonFX(Constants.Motors.shooterID);
     private final Servo shooterServo = new Servo(Constants.Motors.shooterServoID);
     public static final double shooterSpin = 1.0;
     public static final int motorStop = 0;
     public static final double turretPower = 0.1;
+    private final SimpleMotorFeedforward shooterFeed = new SimpleMotorFeedforward(
+        Constants.ShooterPID.kSVolts, Constants.ShooterPID.kVVoltSecondsPerRotation);
 
+    /**
+     * Create shooter class for PID
+     */
     public Shooter() {
+        super(new PIDController(Constants.ShooterPID.kP, Constants.ShooterPID.kI,
+            Constants.ShooterPID.kD));
         shooter.setInverted(true);
+        shooter.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor, 1, 1);
+        getController().setTolerance(50);
+        setSetpoint(85);
+    }
+
+    @Override
+    public void useOutput(double output, double setpoint) {
+        shooter.setVoltage(output + shooterFeed.calculate(setpoint));
+    }
+
+    @Override
+    public double getMeasurement() {
+        double selSenVel = shooter.getSelectedSensorVelocity(0);
+
+        double rotPerSec = (double) selSenVel / Constants.ShooterPID.kUnitsPerRevolution
+            * 10; /* scale per100ms to perSecond */
+        return rotPerSec;
+        // double rotPerMin = RotPerSec * 60.0;
+        // return rotPerMin;
+    }
+
+    @Override
+    public void periodic() {
+        if (m_enabled) {
+            useOutput(m_controller.calculate(getMeasurement(), getSetpoint()), getSetpoint());
+            double selSenVel = shooter.getSelectedSensorVelocity(0);
+            double rotPerSec = (double) selSenVel / Constants.ShooterPID.kUnitsPerRevolution
+                * 10; /* scale per100ms to perSecond */
+            double rotPerMin = rotPerSec * 60.0;
+
+            System.out.println("RPM (Speed): " + rotPerMin);
+            System.out.println("Voltage: " + shooter.getMotorOutputVoltage());
```

Comment:
delete this.  Move the print statements to the `getMeasurement` function to consolidate and comment out the print statements as they slow the program down. We can always uncomment when testing.